sound/soc/sof/ipc4-topology.c

```diff
@@ -548,6 +549,10 @@ static int sof_ipc4_widget_setup_comp_dai(struct snd_sof_widget *swidget)
 	if (ret)
 		goto free_copier_config;
 
+	/* set DAI ops */
+	if (sof_ops(sdev)->select_dai_widget_ops)
+		sof_ops(sdev)->select_dai_widget_ops(sdev, swidget);
```

Comment:
Can't you do this when the PCM is opened based on the same information in Intel code? Or when first time you need this information in Intel code and it is not set yet?